### PR TITLE
Preloader Added

### PR DESCRIPTION
**Pull request by YashBawa**
Fixed Issue #48
Preloader Added by adding code to Html and Css.
No other files were made changes . 
The preloader is working properly, animation created using Vanilla CSS
Reference of the work 
before:
image
after 
image
image
Animation is working well , to get a better view please locally.
The color Theme of the preloader matches with the color theme of website.

### CODE REVIEW

**Comment by YashBawa** [comment]
@matteobruni please review and verify it.
Thanks!!

File: index.html [hunk 1]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
     </div>
   </main>
 
+  <!-- PRELOADER JS-->
```
Comment (matteobruni) [inline index.html:71]: Please remove comments from the HTML

File: index.html [hunk 2]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
...
+  <!-- preloader script tag -->
```
Comment (matteobruni) [inline index.html:72]: Please remove comments from the HTML

File: index.html [hunk 3]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
...
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
+  <!--preloader JS -->
```
Comment (matteobruni) [inline index.html:74]: Please remove comments from the HTML

File: index.html [hunk 4]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
     </div>
   </main>
 
+  <!-- PRELOADER JS-->
+  <!-- preloader script tag -->
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
```
Comment (matteobruni) [inline index.html:73]: jQuery is already included below, please remove this

File: index.html [hunk 5]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
...
+  <!--preloader JS -->
+  <script>
```
Comment (matteobruni) [inline index.html:75]: Move this JavaScript code, without comments, in the `js/main.js` file, or create a `js/preloader.js` file and include it in the page.

File: index.html [hunk 6]
```diff
@@ -65,6 +68,22 @@ <h1>tsParticles 404 Templates</h1>
...
+      $(document).ready(function() {
+      //Preloader
+      preloaderFadeOutTime = 3000;
+      function hidePreloader() {
+      var preloader = $('.loader_bg');
```
Comment (matteobruni) [inline index.html:80]: This code is not properly formatted, please fix it.

**Review by matteobruni — CHANGES_REQUESTED** [review]
I've added some notes to the code, please review all of them

**Comment by YashBawa** [comment]
@matteobruni  I have commited with new changes please check and tell if any more issues needs to be resolved.

File: js/main.js [hunk 7]
```diff
@@ -66,4 +66,15 @@ $(document).ready(function () {
             pages.append(page);
         });
     });
+});
+
+$(document).ready(function()
```
Comment (matteobruni) [inline js/main.js:71]: There’s already a document ready function, please merge them together

**Review by matteobruni — CHANGES_REQUESTED** [review]
Still one change needed

**Comment by YashBawa** [comment]
@matteobruni fixed all issues hope it's fine, if any other changes are needed then please notify!!

**Review by matteobruni — CHANGES_REQUESTED** [review]
I've seen that the preloaded has always 3 seconds of fadeOut, but it should be hidden once the AJAX request is completed. The AJAX request could take milliseconds or seconds, depending on the user browser and connection speed.
The animation is good anyway.

**Comment by YashBawa** [comment]
Keeping a fixed time would be better, because as you mentioned if it takes only milliseconds then the preloader would not look proper and would look more like just an added page for a short while.

**Comment by matteobruni** [comment]
> Keeping a fixed time would be better, because as you mentioned if it takes only milliseconds then the preloader would not look proper and would look more like just an added page for a short while.

image

I need 31ms seconds to complete the AJAX request, waiting the full 3 seconds is a waste of time. It's a simple site that loads a JSON file, the loader should be visible only while that request is still processing.

**Comment by YashBawa** [comment]
Ok will add according

**Pull request closed without merging**